### PR TITLE
[source-bing-ads] add reports_end_date field to Bing Ads connector

### PR DESCRIPTION
**Pull request by vytautasnord**
Issue raised: https://github.com/airbytehq/airbyte/issues/33351

In this pull request, the option for an end date in reports has been implemented. Now, it's possible to resynchronize data for a specific period without the need to load all data until the current date.

The implemented reports_end_date field is optional, and the main change is as follows:
**Previously:**
`report_time.CustomDateRangeEnd = self.get_request_date(reporting_service, datetime.utcnow())`
**Now:**
`report_time.CustomDateRangeEnd = self.get_request_date(reporting_service, end_date)`, where end_date is the date specified in the specifications, or datetime.utcnow() if not defined.

### CODE REVIEW

**Comment by marcosmarxm** [comment]
Do you have a link to Bing documentation showing the option of end date for the reports endpoints?
Is available to all reports and custom ones?

**Comment by vytautasnord** [comment]
@marcosmarxm Documentation link with End Date: https://learn.microsoft.com/en-us/advertising/reporting-service/reporttime?view=bingads-13.

The end field is currently implemented in the current Bing Ads connector. However, its value is set to datetime.utcnow(). Therefore, I have made a change to retrieve this value from the source configuration.

**Comment by vytautasnord** [comment]
@marcosmarxm any update about this one?

**Review by artem1205 — CHANGES_REQUESTED** [review]

**Comment by marcosmarxm** [comment]
@vytautasnord, could you please check the last review comment?

File: airbyte-integrations/connectors/source-bing-ads/source_bing_ads/report_streams.py [hunk 1]
```diff
@@ -161,6 +162,13 @@ def get_start_date(self, stream_state: Mapping[str, Any] = None, account_id: str
                 return pendulum.parse(stream_state[account_id][self.cursor_field])
 
         return self.client.reports_start_date
+    
+    def get_end_date(self, stream_state: Mapping[str, Any] = None, account_id: str = None):
+        if stream_state and account_id:
+            if stream_state.get(account_id, {}).get(self.cursor_field):
+                return pendulum.from_timestamp(stream_state[account_id][self.cursor_field])
```
[the remark this answers is not in the document]
Comment (vytautasnord) [reply airbyte-integrations/connectors/source-bing-ads/source_bing_ads/report_streams.py]: Hi @artem1205 ,
Sorry for the late response. This `get_end_date` function isn't really necessary here. It's sufficient to set the end_date from `client.reports_end_date` to run the report for a specific date range.

[the remark this answers is not in the document]
Comment (vytautasnord) [reply airbyte-integrations/connectors/source-bing-ads/source_bing_ads/report_streams.py]: @artem1205 @marcosmarxm Could you please review this PR again?

**Comment by vytautasnord** [comment]
Hi @marcosmarxm  @artem1205 

I hope you're doing well. Could you please take a look at this MR again when you have a moment? Recently, I encountered an issue where I needed to extract BingAds data for a specific period, and the reports_end_date functionality would have been really helpful.

Thank you in advance for your time and assistance!

**Comment by tautvydas-v** [comment]
@marcosmarxm Can this be reviewed once again? I see that the MR should work, at least in our use case, so not sure what's the progress here

**Pull request closed without merging**